docs/core/reference/versioning.md

```diff
@@ -0,0 +1,72 @@
+# Chain Core Versioning
+
+## Summary
+
+Chain Core versioning falls into two categories: **package** versioning and **network** versioning.
+
+Package versioning applies to the pieces of software that work together to participate in a blockchain network. For example, the Chain Core Mac app and the Java SDK.
+
+Network versioning applies to the network API - the interface through which different Chain Cores on a network comunicate to exchange transactions and blocks. Any breaking changes to the network API interface or Chain protocol are reflected by a change to the network version.
+
+## Package Versioning
+The various packages of Chain Core each use a three number versioning scheme - **x.y.z**. The first two numbers indicate compatablity between packages. For example, if you are a running a version of the Chain Core Mac app whose first two numbers are **1.1**, then you should also use a version of an SDK whose first two numbers are also **1.1**.
+
+Sometimes we make bugfixes and minor feature updates to individual software packages. When this occurs, we update the third number in the version string for that package only. This number doesn't affect compatibility with other packages. For example, you can safely use version **1.1.1** of the Chain Core Mac App with version **1.1.3** of the Java SDK.
+
+### Scope
+
+The package versioning scheme covers a suite of individual software packages that make up Chain Core, including:
+
+- `cored`, the Chain Core server daemon
+- `corectl`, a CLI utility for operating Chain Core
+- `signerd`, the HSM signing server daemon
+- HSM firmware
```

Comment:
Perhaps we should treat the HSM firmware as a whole separate "product" and give it its own version sequence. Updates to it should be rare.

We probably don't want to be making people upgrade HSM firmware just because they went from cored 1.4 to cored 1.5. Both because that's probably unnecessary most of the time, and because touching the HSM is relatively difficult and expensive compared to deploying things on normal servers.

I'm not super confident of this, it's just my first instinct.